src/main/scala/firrtl/transforms/SimpleArithmetic.scala

```diff
@@ -0,0 +1,201 @@
+// See LICENSE for license details.
+
+package firrtl.transforms
+
+import firrtl.Mappers._
+import firrtl.PrimOps._
+import firrtl.traversals.Foreachers._
+import firrtl.Utils.isSimpleArithmeticExpr
+import firrtl._
+import firrtl.WrappedExpression.we
+import firrtl.analyses.InstanceGraph
+import firrtl.ir._
+
+import scala.collection.mutable
+
+object SimpleArithmetic {
+  // Checks if an Expression is made up of only expressions terminated by a Literal or Reference.
+  // private because it's not clear if this definition of "Simple Expression" would be useful elsewhere.
+  // Note that this can have false negatives but MUST NOT have false positives.
+  private def isSimpleExpr(expr: Expression): Boolean = expr match {
+    case _: WRef | _: Literal | _: WSubField => true
+    case DoPrim(op, args, _, _) if isSimpleArithmeticExpr(op) => args.forall(isSimpleExpr)
+    case _ => false
+  }
+
+  def notMax(n: BigInt, w: BigInt): Boolean = {
+    val n1 = -(BigInt(1) << (w.toInt-1)) - 1
+    n != n1
+  }
+
+  /**
+    *
+    * @param e
+    * @return
+    */
+  def replaceSimple(e: Expression): Expression = {
+    e match {
+      case DoPrim(Add, Seq(e, lit@SIntLiteral(n, _)), consts: Seq[BigInt], tpe) if (n < 0) =>  DoPrim(Sub, Seq(e, lit.copy(n * -1)), consts, tpe)
+      case DoPrim(Addw, Seq(e, lit@SIntLiteral(n, _)), consts: Seq[BigInt], tpe) if (n < 0) =>  DoPrim(Subw, Seq(e, lit.copy(n * -1)), consts, tpe)
+      case DoPrim(Sub, Seq(e, lit@SIntLiteral(n, IntWidth(w))), consts: Seq[BigInt], tpe) if (n < 0) && notMax(n, w) =>
+        DoPrim(Add, Seq(e, lit.copy(n * -1)), consts, tpe)
+      case DoPrim(Subw, Seq(e, lit@SIntLiteral(n, IntWidth(w))), consts: Seq[BigInt], tpe) if (n < 0) && notMax(n, w) =>
+        DoPrim(Addw, Seq(e, lit.copy(n * -1)), consts, tpe)
+      case _ => e
+    }
+  }
```

Comment:
When matching on `consts`, you don't need to specify the type `Seq[BigInt]` because it's always that type.